app/views/onboarding/Welcome.tsx

```diff
@@ -0,0 +1,247 @@
+    getUserLocaleOverride().then((locale) => {
+      if (locale) {
+        setLocale(locale);
+      }
+    });
+  });
+
+  const toggleCheckbox = () => {
+    switch (eulaAgreement) {
+      case 'Agreed': {
+        setEulaAgreement('NotAgreed');
+        break;
+      }
+      case 'NotAgreed': {
+        setEulaAgreement('Agreed');
+        break;
+      }
+    }
+  };
+
+  const onLocaleChange = async (locale: string) => {
+    if (locale) {
+      try {
+        await setUserLocaleOverride(locale);
+        setLocale(locale);
+      } catch (err) {
+        console.log('Something went wrong in language change', err);
+      }
+    }
+  };
+
+  const handleOnPressGetStarted = () => {
+    if (eulaAgreement === 'Agreed') {
+      navigation.navigate('Onboarding2');
+    }
+  };
+
+  const handleOnCloseModal = () => {
+    setModalVisible(false);
+  };
+
+  const handleOnPressTermsOfUse = () => {
+    setModalVisible(true);
+  };
+
+  const showModal = modalVisible;
+  const isDisabled = eulaAgreement !== 'Agreed';
+  const isChecked = eulaAgreement === 'Agreed';
+  const checkboxIcon = isChecked
+    ? Images.BoxCheckedIcon
+    : Images.BoxUncheckedIcon;
+  const buttonStyle = isDisabled ? styles.disabledButton : styles.enabledButton;
+  const buttonTextStyle = isDisabled
+    ? styles.disabledButtonText
+    : styles.enabledButtonText;
+
+  const LanguageSelector = () => {
+    return (
+      <NativePicker
+        items={LOCALE_LIST}
+        value={locale}
+        onValueChange={onLocaleChange}>
+        {({ label, openPicker }: { label: string; openPicker: () => void }) => (
+          <TouchableOpacity onPress={openPicker} style={styles.languageButton}>
+            <Typography style={styles.languageSelectorText}>{label}</Typography>
+          </TouchableOpacity>
+        )}
+      </NativePicker>
+    );
+  };
+
+  return (
+    <ImageBackground
+      source={Images.LaunchScreenBackground}
+      style={styles.backgroundImage}>
+      <ImageBackground
+        source={Images.LaunchScreenBackgroundOverlay}
+        style={styles.backgroundImage}>
+        <View style={styles.container}>
+          <View style={styles.headerContainer}>
+            <LanguageSelector />
+          </View>
+
+          <View style={styles.bodyContainer}>
+            <Typography style={styles.mainText}>
+              {t('label.launch_screen1_header')}
+            </Typography>
+          </View>
+
+          <View style={styles.footerContainer}>
+            <View style={styles.checkboxContainer}>
+              <TouchableOpacity
+                testID={'welcome-eula-checkbox'}
+                style={styles.checkbox}
+                onPress={toggleCheckbox}
+                accessible
+                accessibilityRole='checkbox'
+                accessibilityLabel={t('onboarding.eula_message')}>
+                <Image source={checkboxIcon} style={styles.checkboxIcon} />
+              </TouchableOpacity>
+
+              <View>
```

Comment:
I wouldn't necessarily change anything here because I think it's just a downside of React, but to my eye this becomes really hard to read once we have all this nesting and indentation.